python_modules/libraries/dagster-databricks/dagster_databricks_tests/test_databricks_pyspark_step_launcher.py

```diff
@@ -0,0 +1,69 @@
+import os
+
+import pytest
+from dagster_databricks.databricks_pyspark_step_launcher import (
+    DAGSTER_SYSTEM_ENV_VARS,
+    DatabricksPySparkStepLauncher,
+)
+
+
+@pytest.fixture
+def mock_step_launcher_factory():
+    def _mocked(add_dagster_env_variables: bool, env_variables: dict):
+        return DatabricksPySparkStepLauncher(
+            run_config={"some": "config"},
+            permissions={"some": "permissions"},
+            databricks_host="databricks.host.com",
+            databricks_token="abc123",
+            secrets_to_env_variables=[{"some": "secret"}],
+            staging_prefix="/a/prefix",
+            wait_for_logs=False,
+            max_completion_wait_time_seconds=100,
+            env_variables=env_variables,
+            add_dagster_env_variables=add_dagster_env_variables,
+            local_dagster_job_package_path="some/local/path",
+        )
+
+    return _mocked
+
+
+class TestCreateRemoteConfig:
+    def test_given_add_dagster_env_vars_retrieves_dagster_system_vars(
+        self, mock_step_launcher_factory, monkeypatch
+    ):
+        test_env_variables = {"add": "this"}
+        test_launcher = mock_step_launcher_factory(
+            add_dagster_env_variables=True, env_variables=test_env_variables
+        )
+        system_vars = {}
+        for var in DAGSTER_SYSTEM_ENV_VARS:
+            system_vars[var] = f"{var}_value"
+            monkeypatch.setenv(var, f"{var}_value")
+
+        correct_vars = dict(**system_vars, **test_env_variables)
+        env_vars = test_launcher.create_remote_config()
+        assert env_vars.env_variables == correct_vars
+
+    def test_given_no_add_dagster_env_vars_no_system_vars_added(
+        self, mock_step_launcher_factory, monkeypatch
+    ):
+        vars_to_add = {"add": "this"}
+        test_launcher = mock_step_launcher_factory(
+            add_dagster_env_variables=False, env_variables=vars_to_add
+        )
+        for var in enumerate(DAGSTER_SYSTEM_ENV_VARS):
```

Comment:
Hm... I think this line is throwing an error. I think we probably need to take out the `enumerate` call?
```
[2022-11-14T22:42:05Z] =================================== FAILURES ===================================
--
  | [2022-11-14T22:42:05Z] _ TestCreateRemoteConfig.test_given_no_add_dagster_env_vars_no_system_vars_added _
  | [2022-11-14T22:42:05Z]
  | [2022-11-14T22:42:05Z] self = <dagster_databricks_tests.test_databricks_pyspark_step_launcher.TestCreateRemoteConfig object at 0x7fe234f219a0>
  | [2022-11-14T22:42:05Z] mock_step_launcher_factory = <function mock_step_launcher_factory.<locals>._mocked at 0x7fe0ae9f4ee0>
  | [2022-11-14T22:42:05Z] monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7fe0ae9aa520>
  | [2022-11-14T22:42:05Z]
  | [2022-11-14T22:42:05Z]     def test_given_no_add_dagster_env_vars_no_system_vars_added(
  | [2022-11-14T22:42:05Z]         self, mock_step_launcher_factory, monkeypatch
  | [2022-11-14T22:42:05Z]     ):
  | [2022-11-14T22:42:05Z]         vars_to_add = {"add": "this"}
  | [2022-11-14T22:42:05Z]         test_launcher = mock_step_launcher_factory(
  | [2022-11-14T22:42:05Z]             add_dagster_env_variables=False, env_variables=vars_to_add
  | [2022-11-14T22:42:05Z]         )
  | [2022-11-14T22:42:05Z]         for var in enumerate(DAGSTER_SYSTEM_ENV_VARS):
  | [2022-11-14T22:42:05Z] >           monkeypatch.setenv(var, f"{var}_value")
  | [2022-11-14T22:42:05Z]
  | [2022-11-14T22:42:05Z] dagster_databricks_tests/test_databricks_pyspark_step_launcher.py:55:
  | [2022-11-14T22:42:05Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  | [2022-11-14T22:42:05Z] .tox/py39/lib/python3.9/site-packages/_pytest/monkeypatch.py:302: in setenv
  | [2022-11-14T22:42:05Z]     self.setitem(os.environ, name, value)
  | [2022-11-14T22:42:05Z] .tox/py39/lib/python3.9/site-packages/_pytest/monkeypatch.py:266: in setitem
  | [2022-11-14T22:42:05Z]     self._setitem.append((dic, name, dic.get(name, notset)))
  | [2022-11-14T22:42:05Z] /usr/local/lib/python3.9/_collections_abc.py:763: in get
  | [2022-11-14T22:42:05Z]     return self[key]
  | [2022-11-14T22:42:05Z] /usr/local/lib/python3.9/os.py:676: in __getitem__
  | [2022-11-14T22:42:05Z]     value = self._data[self.encodekey(key)]
  | [2022-11-14T22:42:05Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  | [2022-11-14T22:42:05Z]
  | [2022-11-14T22:42:05Z] value = (0, 'DAGSTER_CLOUD_DEPLOYMENT_NAME')
  | [2022-11-14T22:42:05Z]
  | [2022-11-14T22:42:05Z]     def encode(value):
  | [2022-11-14T22:42:05Z]         if not isinstance(value, str):
  | [2022-11-14T22:42:05Z] >           raise TypeError("str expected, not %s" % type(value).__name__)
  | [2022-11-14T22:42:05Z] E           TypeError: str expected, not tuple
```

Reply:
ugh sorry! I should've ran those tests again after I made those changes.